source/common/secret/secret_manager_impl.cc

```diff
@@ -86,13 +87,26 @@ TlsCertificateConfigProviderSharedPtr SecretManagerImpl::findOrCreateTlsCertific
       removeDynamicSecretProvider(map_key);
     };
 
-    secret_provider = std::make_shared<TlsCertificateSdsApi>(
+    secret_provider = create_fn(unregister_secret_provider);
+    dynamic_secret_providers_[map_key] = secret_provider;
+  }
+  return secret_provider;
+}
+
+TlsCertificateConfigProviderSharedPtr SecretManagerImpl::findOrCreateTlsCertificateProvider(
+    const envoy::api::v2::core::ConfigSource& sds_config_source, const std::string& config_name,
+    Server::Configuration::TransportSocketFactoryContext& secret_provider_context) {
+  std::function<SdsApiSharedPtr(std::function<void()> unregister_secret_provider)> create_fn =
+      [&sds_config_source, &config_name, &secret_provider_context](
+          std::function<void()> unregister_secret_provider) -> SdsApiSharedPtr {
+    return std::make_shared<TlsCertificateSdsApi>(
```

Comment:
How about move this make_shared code to a static create() function in the

TlsCertificateSdsApi class

Reply:
Done.